parse.go

```diff
@@ -694,20 +748,39 @@ func isFlag(s string) bool {
 	return strings.HasPrefix(s, "-") && strings.TrimLeft(s, "-") != ""
 }
 
-// val returns a reflect.Value corresponding to the current value for the
-// given path
-func (p *Parser) val(dest path) reflect.Value {
+// getVal returns a reflect.Value corresponding to the current value for
+// the given path, but does not instantiate nil values in the path.
+func (p *Parser) getVal(dest path) reflect.Value {
 	v := p.roots[dest.root]
 	for _, field := range dest.fields {
 		if v.Kind() == reflect.Ptr {
 			if v.IsNil() {
-				return reflect.Value{}
+				v = reflect.New(v.Type().Elem())
 			}
 			v = v.Elem()
 		}
 
 		v = v.FieldByIndex(field.Index)
 	}
+
+	return v
+}
+
+// setVal returns a reflect.Value corresponding to the current value for the
+// given path initiating nil pointers in the path
+func (p *Parser) setVal(dest path) reflect.Value {
```

Comment:
I see what you're doing here but I think the name `setVal` is a little confusing because it sounds as if it's going to set the value corresponding to the path, whereas in fact you're getting the Value object corresponding to the path while filling in nil pointers. How about calling this `settableVal`

Reply:
I have reverted the naming change and kept the `val(` function name. The improved behavior is still needed since it can be used to set an attribute in a nested option group or sub-command to which the path may consist of nil pointers. In these scenarios the parents of the deeply nested variable needs to be created to avoid nil-pointers.

This also avoids the `v.Set(...)` call to instantiate a sub-command pointer. The sub-command struct pointer is now created just because the command name is mentioned in the arguments.